redisson/src/main/java/org/redisson/RedissonPriorityBlockingQueue.java

```diff
@@ -115,6 +115,9 @@ public V take() throws InterruptedException {
     }
 
     public RFuture<V> pollAsync(long timeout, TimeUnit unit) {
+        if (timeout < 0) {
```

Comment:
This logic should be applied to takeAsync(CompletableFuture<V> result, long delay, long timeoutInMicro, RedisCommand<T> command, Object... params) method only